From 289203dcc6b2b521693d7086d81e8d7c92a86457 Mon Sep 17 00:00:00 2001
From: Bo Fang <bofang1@codeaurora.org>
Date: Wed, 15 Apr 2020 21:44:13 +0800
Subject: [PATCH 22/28] waylandsink: add NV12_UBWC in sink pad cap template

Since waylandsink support NV12_UBWC, should add it into sink
pad cap template. Otherwise, when user explicitly set cap's
format as NV12_UBWC like "gst-launch-1.0 ... ! qvconv ! queue
! video/x-raw, format=NV12_UBWC ! waylandsink", pipeline will
fail.

Change-Id: I2360447386321742715ffe208322323b43e0a11c
Signed-off-by: Bo Fang <bofang1@codeaurora.org>
---
 ext/wayland/gstwaylandsink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index a5e2240..927d06c 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -125,7 +125,7 @@ static inline void gbm_swframe_mem_destroy(swfrm_gbm_item* item)
 #define WL_VIDEO_FORMATS \
     "{ BGRx, BGRA, RGBx, xBGR, xRGB, RGBA, ABGR, ARGB, RGB, BGR, " \
     "RGB16, BGR16, YUY2, YVYU, UYVY, AYUV, NV12, NV21, NV16, " \
-    "YUV9, YVU9, Y41B, I420, YV12, Y42B, v308 }"
+    "YUV9, YVU9, Y41B, I420, YV12, Y42B, v308, NV12_UBWC }"
 
 static GstStaticPadTemplate sink_template = GST_STATIC_PAD_TEMPLATE ("sink",
     GST_PAD_SINK,
-- 
2.7.4

